plugins/topology.c

```diff
@@ -321,76 +296,22 @@ struct listchannels_opts {
 	struct short_channel_id *scid;
 };
 
-/* We record which local channels are valid; we could record which are
- * invalid, but our testsuite has some weirdness where it has local
- * channels in the store it knows nothing about. */
-static struct node_map *local_connected(const tal_t *ctx,
-					const char *buf,
-					const jsmntok_t *result)
-{
-	size_t i;
-	const jsmntok_t *t, *peers = json_get_member(buf, result, "peers");
```

Comment:
I removed this because I was thinking that it was introduced for historical reasons.

However, I'm sure that this change will cause [this failure](https://github.com/ElementsProject/lightning/runs/5501615365?check_suite_focus=true#step:5:8659) because the `getroute` use the `listchannels` command, but I'm not able to find a good reason to introduce the function `local_connected` again

Reply:
No, you will need to restore this!  It can work pretty well off listpeerchannels, no?

Reply:
Right, My original idea was to add a field to the `listpeerchannels` result to avoid an extra caching operation in the `listchannels`.

I'm reintroducing it with a little bit of rework, lets's see what the CI tell us with this last change, on my machine all the test execution is too slow, so sorry if there future errors in the CI